src/toolkits/object_detection/object_detector.cpp

```diff
@@ -522,45 +538,123 @@ variant_map_type object_detector::evaluate(
     result_map.erase(MAP);
   }
 
-  return result_map;
+  return convert_map_to_types(result_map, output_type);
+}
+
+variant_type object_detector::convert_map_to_types(variant_map_type& result_map, std::string output_type){
```

Comment:
Why do we need a mutable reference for `result_map`?

Could we pass `output_type` by (`const`) reference?